src/implementations/vanilla/components/inpage-navigation/inpage-navigation.js

```diff
@@ -332,15 +334,15 @@ export class InpageNavigation {
       onDeactivate: () => this.closeList(),
     });
 
-    if (this.attachClickListener && toggleElement) {
-      toggleElement.addEventListener('click', this.handleClickOnToggler);
+    if (this.attachClickListener && this.toggleElement) {
+      this.toggleElement.addEventListener('click', this.handleClickOnToggler);
```

Comment:
did not look at the whole file but it seems that we are adding event listeners and not removing them in destroy() here